### PR TITLE
Update LoadingButton role

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This updates where the role="status" is included on the LoadingButton component, to help the text read out more consistently.

Recommended here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/26391#issuecomment-881059876

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26391

## Acceptance criteria
- [ ] Role change improves text readout in screen readers without creating new problems

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/loading-button/LoadingButton.jsx
Comment (jbalboni) [inline src/platform/site-wide/loading-button/LoadingButton.jsx:30]: Sort of concerned that I have to ignore this rule, but I'll defer to VSP and the accessibility folks

Comment (jhonnyoddball) [reply src/platform/site-wide/loading-button/LoadingButton.jsx:30]: Have you tried adding the <button> inside a <div>?

Comment (jhonnyoddball) [reply src/platform/site-wide/loading-button/LoadingButton.jsx:30]: @1Copenut Could we have a couple of eyes here?

Comment (1Copenut) [reply src/platform/site-wide/loading-button/LoadingButton.jsx:30]: @jbalboni & @jhonnyoddball I whiffed on this one big time. We should remove the `role="status"` and add `aria-live="polite"` here instead. The button has an implicit role of "button" which is what the linter rightly caught. I had the code base running locally and tried this out briefly in VO with good results. Let's change it and I'll test in staging on the other two screen readers.

**Review by 1Copenut — CHANGES_REQUESTED** [review]

File: src/platform/site-wide/loading-button/LoadingButton.jsx
Comment (jbalboni) [inline src/platform/site-wide/loading-button/LoadingButton.jsx:30]: @jhonnyoddball @1Copenut Updated

**Review by 1Copenut — APPROVED** [review]
LGTM! Will retest in staging after it's merged

**Review by jhonny218 — APPROVED** [review]
LGTM

**Pull request merged**
